src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -307,16 +300,28 @@ public void deletePaymentMethod(final UUID kbAccountId, final UUID kbPaymentMeth
             } catch (final ServiceException e) {
                 throw new PaymentPluginApiException("Unable to retrieve recurring details in Adyen", e);
             }
-
-            if (!recurringDetailList.isEmpty()) {
-                final String token = recurringDetailList.get(recurringDetailList.size() - 1).getRecurringDetailReference();
+            for (final RecurringDetail recurringDetail : recurringDetailList) {
+                final AdyenResponsesRecord formerResponse;
+                try {
+                    formerResponse = dao.getResponse(recurringDetail.getFirstPspReference());
+                } catch (SQLException e) {
+                    throw new PaymentPluginApiException("Unable to retrieve adyen response", e);
+                }
+                final Payment payment;
+                try {
+                    payment = killbillAPI.getPaymentApi().getPayment(UUID.fromString(formerResponse.getKbPaymentId()), true, properties, context);
+                } catch (PaymentApiException e) {
+                    throw new PaymentPluginApiException("Unable to retrieve Payment for externalKey " + recurringDetail.getFirstPspReference(), e);
+                }
+                if (payment == null) {
```

Comment:
`payment == null` should never happen (an exception is thrown in case we couldn't find it).

Also, I think we want to check whether this payment method record maps to that payment (i.e. `record. getKbPaymentMethodId().equals(payment.getPaymentMethodId())`), no?